Single redhat module for splitting auter to support different distributions

reaperzn opened the pull request:
This is a simplification of #32 which only includes the redhat module. This is becoming more difficult to maintain as changes are being made in the master branch before this is merged. The only change here is that the prepare_updates and apply_updates have been moved to a separate script. This means that in future, we can build more modules for other distributions if required.

Commit:
Single redhat module for splitting auter to support different distributions

piersc commented:
- Lets call this "yumdnf" module, since the package manager is what matters most.
- The modules aren't runnable themselves so shouldn't have the shebang header
- The functions get included, so simplify by just calling those without args
- I don't think this comes from your PR, but the following is hard coded to yum, I assume this just redirects to dnf, but its still not right:
 local TRANSACTIONID=$(yum history info 2>&1 | grep "Transaction ID")
- Remove the apt-get reference since this is just for dnf/yum right now
- readonly FUNCTIONDIR="/usr/lib/auter" no need to define this is we'll only use once

Beerey commented:
Can we add some safety code to stop the modules being run directly, since if they are they'll be missing lots of variables and will have undefined results?  Something like this:

http://stackoverflow.com/questions/2683279/how-to-detect-if-a-script-is-being-sourced

Beerey commented:
Also the change notes etc need to be updated if you're going to release a new version since there were a few other changes.

Commit:
- renamed auter.redhatModule to auter.yumdnfModule
- removed #! from auter.yumdnfModule
- updated spec file for new name
- fixed TRANSACTIONID declaration to use ${PACKAGE_MANAGER}
- Removed references to apt-get
- removed the FUNCTIONDIR variable
- moved the sourcing to before the functions are called and removed the locally declared functions with the same names
- removed the argument requirement from auter.yumdnfModule
- Added a line to ensure the script is not executed directlry by a user

Commit:
Removed the entry from the changelog in the spec file

piersc commented:
```+# Source the module for the specific OS distribution```

the module is per package manager, not distro. Same with the header comment in auter.yumdnfModule


The following spec file changes are needed. Can't use the macro for name since the file is explicitly sourced:

```
< install -p -m 0755 %{name}.yumdnfModule %{buildroot}%{_usr}/lib/%{name}/auter.module
---
> install -p -m 0755 %{name}.yumdnfModule %{buildroot}%{_usr}/lib/%{name}.module
81d80
< %{_usr}/lib/%{name}/auter.module
91a91
> %{_usr}/lib/%{name}.module
```

Commit:
Corrected wording in auter and mistakes in auter.spec files

piersc commented:
Fix up the header comment in auter.yumdnfModule too

Commit:
Reworded the description of this file to specify this is the yum/dnf modue

piersc commented:
Merged, thanks!
